scripts/qa_crowbarsetup.sh

```diff
@@ -3232,53 +3232,71 @@ function oncontroller_manila_generic_driver_setup()
-    if [[ $ret =~ "already there; not retrieving" ]]; then
+    local ret=$(wget -nc -O $service_image_name "$service_image_url" 2>&1 >/dev/null)
+    if [[ $ret =~ "already there; not retrieving" ]] || [[ $ret =~ "200 OK" ]]; then
         echo $ret
     elif [[ $ret =~ "Not Found" ]]; then
-        complain 73 "manila image not found"
+        complain 73 "manila image not found: $ret"
+    else
+        complain 74 "failed to retrieve manila image: $ret"
     fi
 
     . .openrc
-    manila_service_tenant_id=`openstack project create manila-service -f value -c id`
-    openstack role add --project manila-service --user admin admin
-    export OS_TENANT_NAME='manila-service'
-    openstack image create --file $service_image_name \
-        $service_image_params --container-format bare --public \
-        manila-service-image
-    nova flavor-create manila-service-image-flavor 100 512 0 1
-
-    nova secgroup-create $sec_group "$sec_group description"
-    nova secgroup-add-rule $sec_group icmp -1 -1 0.0.0.0/0
-    nova secgroup-add-rule $sec_group tcp 22 22 0.0.0.0/0
-    nova secgroup-add-rule $sec_group tcp 2049 2049 0.0.0.0/0
-    nova secgroup-add-rule $sec_group tcp 20000 65535 0.0.0.0/0
-    nova secgroup-add-rule $sec_group udp 2049 2049 0.0.0.0/0
-    nova secgroup-add-rule $sec_group udp 445 445 0.0.0.0/0
-    nova secgroup-add-rule $sec_group tcp 445 445 0.0.0.0/0
-    nova secgroup-add-rule $sec_group tcp 137 139 0.0.0.0/0
-    nova secgroup-add-rule $sec_group udp 137 139 0.0.0.0/0
-    nova secgroup-add-rule $sec_group tcp 111 111 0.0.0.0/0
-    nova secgroup-add-rule $sec_group udp 111 111 0.0.0.0/0
-
-    fixed_net_id=`neutron net-show fixed -f value -c id`
-    timeout 10m nova boot --poll --flavor 100 --image manila-service-image \
-        --security-groups $sec_group,default \
-        --nic net-id=$fixed_net_id manila-service
-
-    [ $? != 0 ] && complain 43 "nova boot for manila failed"
-
-    # manila tempest tests use the floating IP to export the shares. So create
-    # a floating IP for the manila instance and add it to the VM
-    manila_tenant_vm_ip=`openstack ip floating create floating -f value -c ip`
-    openstack ip floating add $manila_tenant_vm_ip manila-service
-
-    [ $? != 0 ] && complain 44 "adding a floating ip to the manila service VM failed"
-
-    # check that the service VM is pingable. otherwise manila tempest tests will fail later
-    wait_for 300 1 "nc -z $manila_tenant_vm_ip 22" \
-        "manila service VM booted and ssh port open" \
-        "echo \"ERROR: manila service VM not listening on ssh port. manila tests will fail!\""
+
+    if ! openstack project show manila-service; then
+        manila_service_tenant_id=`openstack project create manila-service -f value -c id`
+        openstack role add --project manila-service --user admin admin
+        export OS_TENANT_NAME='manila-service'
+    fi
+
+    if ! openstack image list | grep -q manila-service-image; then
```

Comment:
Maybe document that you don't use `openstack image show` because it apparently only works with image IDs, not image names (!!!).

Btw, a stricter check would be: `openstack image list --f value -c Name | grep -q "^manila-service-image$"`